src/OrbitBase/include/OrbitBase/Sort.h

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_BASE_SORT_H_
+#define ORBIT_BASE_SORT_H_
+
+#include <algorithm>
+#include <functional>
+#include <type_traits>
+
+namespace orbit_base_internal {
+
+template <typename Comparator, typename Projection>
+static auto MakeComparator(Comparator comparator, Projection projection) {
+  return [comparator, projection](const auto& a, const auto& b) {
+    return comparator(projection(a), projection(b));
+  };
+}
+
+}  // namespace orbit_base_internal
+
+namespace orbit_base {
+
+template <typename RandomIt, typename Projection, typename Comparator = std::less<>,
```

Comment:
Maybe std::identity as the default argument for Projection?

Reply:
Again, why? If I wish sort with identity Projection, I'll just use std::sort.